simple_calculator/simple_calculator.py

```diff
@@ -0,0 +1,34 @@
+
```

Comment:
Hi @ezems123 Good job. Just some suggestions to the code. How does this look?

```
import argparse

def main():
    parser = argparse.ArgumentParser(description="calculator")
    parser.add_argument('integers', metavar='N', type=int, nargs=2,
                    help='an integer for the accumulator')
    parser.add_argument('--add', action='store_true', help="Adds a set of numbers")
    parser.add_argument('--sub', action='store_true', help="Subtracts a set of numbers")
    parser.add_argument('--multiply', action='store_true', help="Multiplies a set of numbers")
    parser.add_argument('--divide', action='store_true', help="Divides a set of numbers")

    args = parser.parse_args()
    print(calc(args))


def calc(args):
    if args.add:
        return sum(args.integers)
    if args.sub:
        return args.integers[0]-args.integers[1]
    if args.multiply:
        return args.integers[0]*args.integers[1]
    if args.divide:
        return args.integers[0]/args.integers[1]
    else:
        return "You have chosen an invalid opearation"


if __name__ == "__main__":
    main()
```

1. Now, it should be easier to operate on more than 2 numbers. Can you try to do that using the above code?
2. Can you also try to capture exceptions, such as if we divide by 0?
3. Can we do this for decimal values instead of int?

Reply:
Hello @Talina06 ,
1. I am getting a few errors. I need a guide.

![screenshot](https://user-images.githubusercontent.com/35050235/66248653-5fe08f00-e721-11e9-96d8-8a8909252cf6.jpg)

Response awaited.

Reply:
Hello @Talina06  ,
Ran the code on VSCode, result is 100%. I guess it is a thing with python IDLE.
For upto four arguments:

`
import argparse


def main():
    parser = argparse.ArgumentParser(description="calculator")
    parser.add_argument('integers', metavar='N', type=int, nargs=4,
                        help='an integer for the accumulator')
    parser.add_argument('--add', action='store_true',
                        help="Adds a set of numbers")
    parser.add_argument('--sub', action='store_true',
                        help="Subtracts a set of numbers")
    parser.add_argument('--multiply', action='store_true',
                        help="Multiplies a set of numbers")
    parser.add_argument('--divide', action='store_true',
                        help="Divides a set of numbers")

    args = parser.parse_args()
    print(calc(args))


def calc(args):
    if args.add:
        return sum(args.integers)
    if args.sub:
        return args.integers[0]-args.integers[1]-args.integers[2]-args.integers[3]
    if args.multiply:
        return args.integers[0]*args.integers[1]*args.integers[2]*args.integers[3]
    if args.divide:
        return args.integers[0]/args.integers[1]/args.integers[2]/args.integers[3]
    else:
        return "You have chosen an invalid opearation"


if __name__ == "__main__":
    main()
`

This is should take up to 4 arguments

I await your comments and direction.

Moving to 
2.Can you also try to capture exceptions, such as if we divide by 0?
3.Can we do this for decimal values instead of int?

Regards

Ezemonye

Reply:
Hello @Talina06 ,
Ran the code on VSCode, result is 100%. I guess it is a thing with python IDLE.
For upto four arguments:

```

import argparse

def main():
parser = argparse.ArgumentParser(description="calculator")
parser.add_argument('integers', metavar='N', type=int, nargs=4,
help='an integer for the accumulator')
parser.add_argument('--add', action='store_true',
help="Adds a set of numbers")
parser.add_argument('--sub', action='store_true',
help="Subtracts a set of numbers")
parser.add_argument('--multiply', action='store_true',
help="Multiplies a set of numbers")
parser.add_argument('--divide', action='store_true',
help="Divides a set of numbers")

args = parser.parse_args()
print(calc(args))
def calc(args):
if args.add:
return sum(args.integers)
if args.sub:
return args.integers[0]-args.integers[1]-args.integers[2]-args.integers[3]
if args.multiply:
return args.integers[0]*args.integers[1]*args.integers[2]*args.integers[3]
if args.divide:
return args.integers[0]/args.integers[1]/args.integers[2]/args.integers[3]
else:
return "You have chosen an invalid opearation"

if name == "main":
main()

```

Reply:
Hello @Talina06 ,
Ran the code on VSCode, result is 100%. I guess it is a thing with python IDLE.
For up to four arguments:

```
import argparse

def main():
        parser = argparse.ArgumentParser(description="calculator")
        parser.add_argument('integers', metavar='N', type=int, nargs=4, help='an integer for the accumulator')
        parser.add_argument('--add', action='store_true', help="Adds a set of numbers")
        parser.add_argument('--sub', action='store_true', help="Subtracts a set of numbers")
        parser.add_argument('--multiply', action='store_true', help="Multiplies a set of numbers")
        parser.add_argument('--divide', action='store_true', help="Divides a set of numbers")

args = parser.parse_args()
print(calc(args))

def calc(args):
        if args.add:
            return sum(args.integers)
        if args.sub:
            return args.integers[0]-args.integers[1]-args.integers[2]-args.integers[3]
        if args.multiply:
            return args.integers[0]*args.integers[1]*args.integers[2]*args.integers[3]
        if args.divide:
            return args.integers[0]/args.integers[1]/args.integers[2]/args.integers[3]
        else:
            return "You have chosen an invalid opearation"

if name == "main":
main()

```

Your comments please.
Moving to items 2 and 3 before tasks in your last email.

Regards

Ezemonye

Reply:
Hello @Talina06 ,
This is for multiple arguments,

```
import argparse

def main():
        parser = argparse.ArgumentParser(description="calculator")
        parser.add_argument('integers', metavar='N', type=int, nargs=+, help='an integer for the accumulator')
        parser.add_argument('--add', action='store_true', help="Adds a set of numbers")
        parser.add_argument('--sub', action='store_true', help="Subtracts a set of numbers")
        parser.add_argument('--multiply', action='store_true', help="Multiplies a set of numbers")
        parser.add_argument('--divide', action='store_true', help="Divides a set of numbers")

args = parser.parse_args()
print(calc(args))

def calc(args):
        if args.add:
            return sum(args.integers)
        if args.sub:
            return args.integers[0]-args.integers[1]-args.integers[2]-args.integers[3]
        if args.multiply:
            return args.integers[0]*args.integers[1]*args.integers[2]*args.integers[3]
        if args.divide:
            return args.integers[0]/args.integers[1]/args.integers[2]/args.integers[3]
        else:
            return "You have chosen an invalid opearation"

if name == "main":
main()
```
Regards

Ezemonye